icelake/src/transaction.rs

```diff
@@ -14,7 +14,7 @@ use std::time::{SystemTime, UNIX_EPOCH};
 use uuid::Uuid;
 
 /// Operation of a transaction.
-enum Operation {
+pub enum Operation {
```

Comment:
I thin we don't need to expose this? We just need to expose `DataFile` to snapshot summary builder and it can do same thing according to content type.